### PR TITLE
Factorial refactoring, added unit tests

### PR DESCRIPTION
**Pull request by albina-astr**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [ ] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

File: src/main/java/com/thealgorithms/maths/Factorial.java [hunk 1]
```diff
@@ -1,15 +1,6 @@
 package com.thealgorithms.maths;
 
 public class Factorial {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/maths/Factorial.java:3]: I think it does not make sense to have any objects of the class `Factorial` - one can make a _proper_ utility  class of it. What do you think about the following change:
```suggestion
public final class Factorial {
    private Factorial() {
    }
```

File: src/test/java/com/thealgorithms/maths/FactorialTest.java [hunk 2]
```diff
@@ -5,9 +5,19 @@
 import org.junit.jupiter.api.Test;
 
 public class FactorialTest {
+    private static final String EXCEPTION_MESSAGE = "Input number cannot be negative!";
```
Comment (siriak) [inline src/test/java/com/thealgorithms/maths/FactorialTest.java:8]:
```suggestion
    private static final String EXCEPTION_MESSAGE = "Input number cannot be negative";
```
I wouldn't be so emotional in the exception message :)

File: src/main/java/com/thealgorithms/maths/Factorial.java [hunk 3]
```diff
@@ -18,11 +9,12 @@ public static void main(String[] args) {
      */
     public static long factorial(int n) {
         if (n < 0) {
-            throw new IllegalArgumentException("number is negative");
+            throw new IllegalArgumentException("Input number cannot be negative!");
```
Comment (siriak) [inline src/main/java/com/thealgorithms/maths/Factorial.java:12]:
```suggestion
            throw new IllegalArgumentException("Input number cannot be negative");
```

File: src/main/java/com/thealgorithms/maths/Factorial.java (hunk 1 above)
Comment (albina-astr) [reply src/main/java/com/thealgorithms/maths/Factorial.java:3]: Thanks for the suggestion. But I think it is not necessary at all.

File: src/test/java/com/thealgorithms/maths/FactorialTest.java (hunk 2 above)
Comment (albina-astr) [reply src/test/java/com/thealgorithms/maths/FactorialTest.java]: Ok, no emotions next time.

File: src/main/java/com/thealgorithms/maths/Factorial.java (hunk 1 above)
Comment (siriak) [reply src/main/java/com/thealgorithms/maths/Factorial.java:3]: @vil02  While the usage of final itself is good, mostly we don't use it at the moment. I think we can make a transition everywhere at once to keep code consistent

**Review by siriak — APPROVED** [review]

File: src/main/java/com/thealgorithms/maths/Factorial.java (hunk 1 above)
Comment (vil02) [reply src/main/java/com/thealgorithms/maths/Factorial.java:3]: @siriak The main part of this suggestion was to explicitly define the default constructor of `Factorial` as private. The `final` is just a nice small addition.

Comment (siriak) [reply src/main/java/com/thealgorithms/maths/Factorial.java:3]: I see now, I missed this at first. Anyway, feel free to raise a PR with the changes you propose, and we can continue discussion there

Comment (vil02) [reply src/main/java/com/thealgorithms/maths/Factorial.java:3]: cf. #4267.

**Pull request merged**